src/main/java/de/tum/cit/aet/artemis/atlas/repository/CourseCompetencyRepository.java

```diff
@@ -315,6 +315,10 @@ default CourseCompetency findByIdWithLectureUnitsAndExercisesElseThrow(long comp
             """)
     List<CourseCompetency> findAllByExerciseIdWithExerciseLinks(@Param("exerciseId") long exerciseId);
 
+    default List<CourseCompetency> findAllByExerciseId(long exerciseId) {
```

Comment:
Please never do this. You essentially rename a method and additionally hide what additional information is retrieved from the DB.

Either directly use the method you're calling in the return statement (only if you need the exercise links which is not the case from what i understand).
OR add this query to the repository to prevent fetching unnecessary data:

```
@Query("""
            SELECT c
            FROM CourseCompetency c
                LEFT JOIN c.exerciseLinks el
            WHERE el.exercise.id = :exerciseId
            """)
    List<CourseCompetency> findAllByExerciseId(@Param("exerciseId") long exerciseId);
```

Please don't underestimate how much impact this has on our DB.

Reply:
Thanks for the feedback - you're right! I removed the redundant wrapper